search/search-client-connect/src/test/java/io/camunda/search/connect/plugin/PluginRepositoryTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.connect.plugin;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.plugin.search.header.CustomHeader;
+import io.camunda.plugin.search.header.DatabaseCustomHeaderSupplier;
+import io.camunda.zeebe.util.jar.ExternalJarClassLoader;
+import io.camunda.zeebe.util.jar.ExternalJarLoadException;
+import io.camunda.zeebe.util.jar.ExternalJarRepository;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class PluginRepositoryTest {
+
+  @Test
+  void shouldReturnUnmodifiableCollectionOfPlugins() {
+    final var initialPlugins =
+        new LinkedHashMap<String, Class<? extends DatabaseCustomHeaderSupplier>>();
+    initialPlugins.put("PLG1", DatabaseCustomHeaderSupplierTestImpl.class);
+    final var jarRepository = Mockito.mock(ExternalJarRepository.class);
+    final var basePath = Mockito.mock(Path.class);
+    final var repo = new PluginRepository(initialPlugins, jarRepository, basePath);
+
+    final var plugins = repo.getPlugins();
+
+    assertThat(plugins).hasSize(1);
+    assertThatCode(() -> plugins.put("PLG2", DatabaseCustomHeaderSupplierTestImpl.class))
+        .isInstanceOf(UnsupportedOperationException.class);
+  }
+
+  @Test
+  void shouldJustReturnSelfWhenLoadIfConfigsWereNull() {
+    final var repo = new PluginRepository();
+
+    final var repoRef = repo.load(null);
+
+    assertThat(repoRef.getPlugins()).isEmpty();
+    assertThat(repo).isSameAs(repoRef);
+  }
+
+  @Test
+  void shouldJustReturnSelfWhenLoadIfConfigsWereEmpty() {
+    final var repo = new PluginRepository();
+
+    final var repoRef = repo.load(new ArrayList<>());
+
+    assertThat(repoRef.getPlugins()).isEmpty();
+    assertThat(repo).isSameAs(repoRef);
+  }
```

Comment:
:wrench: imo those tests are not useful, the semantics that it returns itself as nice, but even if you break them, would not cause errors.

Reply:
Removed